pylint/checkers/typecheck.py

```diff
@@ -523,6 +525,77 @@ def _emit_no_member(
     return True
 
 
+def _get_all_attribute_assignments(
+    node: astroid.FunctionDef, name: str | None = None
+) -> set[str]:
+    attributes = set()
+    for child in node.nodes_of_class((astroid.Assign, astroid.AnnAssign)):
+        targets = []
+        if isinstance(child, astroid.Assign):
+            targets = child.targets
+        elif isinstance(child, astroid.AnnAssign):
+            targets = [child.target]
+        for assign_target in targets:
+            if not isinstance(assign_target, astroid.AssignAttr):
+                continue
+            if not isinstance(assign_target.expr, astroid.Name):
+                continue
```

Comment:
This is uncovered. Can we cover it?

Reply:
Astroid docs says that [AssignAttr.expr is  `Optional[NodeNG]`](https://pylint.pycqa.org/projects/astroid/en/latest/api/astroid.nodes.html#astroid.nodes.AssignAttr.expr), hence the check.
This was more a "defensive coding" thing because it could happen in the wild but I don't know how to trigger it.

Reply:
Let's test this on L546. I highly suspect this is due to incorrect typing in `astroid` (of which there is a lot for node attributes), so it is likely we will never be able to cover this.

Reply:
Sure thing